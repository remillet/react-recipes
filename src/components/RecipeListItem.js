import React from 'react';
import PropTypes from 'prop-types';

const RecipeListItem = (props) => {
  const {
    recipe, favorites, onClick, onFavorited,
  } = props;

  return (
    <li
      className="py1 border-bottom border-bottom-dashed pointer"
      key={recipe.id}
      onClick={() => {
        if (onClick !== null) {
          onClick(recipe.id);
        }
      }}
    >
      <span
        role="img"
        aria-label="favorite"
        className="mr1"
        onClick={
          (e) => {
            e.stopPropagation(); // Prevent the click from bubbling up.
            onFavorited(recipe.id);
          }
        }
      >
        { !favorites.includes(recipe.id) ? '⌑' : '⭐'}
      </span>
      <span>{recipe.name}</span>
      <span> - </span>
      <span>{recipe.category}</span>
    </li>
  );
};

RecipeListItem.defaultProps = {
  recipe: {
    id: -1,
  },
  favorites: [],
  onClick: null,
  onFavorited: null,
};

RecipeListItem.propTypes = {
  recipe: PropTypes.object,
  favorites: PropTypes.arrayOf(String),
  onClick: PropTypes.func,
  onFavorited: PropTypes.func,
};

export default RecipeListItem;
