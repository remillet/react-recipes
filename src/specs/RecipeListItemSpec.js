import React from 'react';
import renderer from 'react-test-renderer';
import RecipeListItem from '../components/RecipeListItem';

describe('<RecipeListItem />', () => {
  test('Should not break if no recipe/details passed in', () => {
    const component = renderer.create(<RecipeListItem />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should correctly render recipe', () => {
  });

  test('Should render favorite state', () => {
  });
});
