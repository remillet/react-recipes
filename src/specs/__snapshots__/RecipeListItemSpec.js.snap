// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeListItem /> Should not break if no recipe/details passed in 1`] = `
<li
  className="py1 border-bottom border-bottom-dashed pointer"
  onClick={[Function]}
>
  <span
    aria-label="favorite"
    className="mr1"
    onClick={[Function]}
    role="img"
  >
    âŒ‘
  </span>
  <span />
  <span>
     - 
  </span>
  <span />
</li>
`;
